src/main/java/nl/mpdev/backend_spring_boot_tech_it_easy_controller/controllers/TelevisionsController.java

```diff
@@ -0,0 +1,78 @@
+package nl.mpdev.backend_spring_boot_tech_it_easy_controller.controllers;
+
+import nl.mpdev.backend_spring_boot_tech_it_easy_controller.exceptions.RecordNotFoundException;
+import nl.mpdev.backend_spring_boot_tech_it_easy_controller.exceptions.StringTooLongException;
+import nl.mpdev.backend_spring_boot_tech_it_easy_controller.models.Television;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/v1")
+public class TelevisionsController {
+
+  List<Television> televisionDataBase = new ArrayList<>();
+
+  @GetMapping("/televisions")
+  public ResponseEntity<List<Television>> getAllTelevisions() {
+//    Map<String, String> response = new HashMap<>();
+//    response.put("television 1","Toshiba");
+//    response.put("television 2", "LG");
+    return ResponseEntity.ok(televisionDataBase);
+  }
+
+  @GetMapping("/televisions/{id}")
+  public ResponseEntity<Television> getTelevisionById(@PathVariable("id") int id) {
+    if (findTelevisionById(id) == null) {
+      throw new RecordNotFoundException();
+    }
+    return ResponseEntity.ok(findTelevisionById(id));
+  }
+
+  @PostMapping("/televisions")
+  public ResponseEntity<Integer> postTelevision(@RequestBody String television) {
+    if (television.length() > 20) {
+      throw new StringTooLongException("This string is way too long");
+    }
+    Television newTV = new Television(television);
+    televisionDataBase.add(newTV);
+    return ResponseEntity.status(HttpStatus.CREATED).body(newTV.getId());
+  }
+
+  @PutMapping("/televisions/{id}")
+  public ResponseEntity<Integer> putTelevision(@PathVariable int id, @RequestBody String television) {
+    if (television.length() > 20) {
+      throw new StringTooLongException("This string is way too long");
+    }
+    else if (findTelevisionById(id) == null) {

```

Comment:
Hier ga je functionaliteiten door elkaar halen. Hier voer je in de PutMapping de functionaliteit van de PostMapping uit. Dat is niet goed. In een PutMapping wil je enkel PutMapping functionaliteiten uitvoeren. Als er in dit geval geen bestaande television gevonden wordt, dan kun je de PutMapping functionaliteit niet uitvoeren en return je dus een 404 naar de gebruiker.